### PR TITLE
open login page when api container is ready

### PR DESCRIPTION
**Pull request by axelchauvin**
use healthcheck endpoint to only open browser when api is ready

### CODE REVIEW

File: deploy-retool [hunk 1]
```diff
@@ -242,13 +246,25 @@ fi
...
+    if ! api_container_up; then
         $MAYBE_SUDO docker-compose up -d > /dev/null 2>&1
     fi
     WAITED=$((WAITED + 1))
     sleep 1
 done
 
+WAITED=0
+# one request every three seconds. It takes about 40 seconds for the api to get ready so 1 minute should be enough margin
+while [ $WAITED -lt 20 ] && ! api_running; do
```
Comment (blackentropy) [inline deploy-retool:258]: Might need to wait longer than 60s since we don't know how fast our customer's computers are.

File: deploy-retool [hunk 2]
```diff
@@ -218,22 +218,26 @@ echo " -- ${GREEN}${BOLD}!! RETOOL IS BOOTING !!${NORMAL} --"
...
-api_running() {
-  API_DOCKER_CONTAINER="$($MAYBE_SUDO docker ps | grep 'retool-onpremise_api')"
+api_container_up() {
+  API_DOCKER_CONTAINER = "$($MAYBE_SUDO docker ps | grep 'retool-onpremise-api')"
   test -n "$API_DOCKER_CONTAINER"
 }
 
+api_running() {
+  healthcheck_response = $(curl -s -o /dev/null -w "%{http_code}" http://localhost:3000/api/checkHealth)
+  test $healthcheck_response -eq 200
```
Comment (blackentropy) [inline deploy-retool:228]: makes sense to me!

File: deploy-retool [hunk 3]
```diff
@@ -218,22 +218,26 @@ echo " -- ${GREEN}${BOLD}!! RETOOL IS BOOTING !!${NORMAL} --"
 
 WAITED=0
 
-api_running() {
-  API_DOCKER_CONTAINER="$($MAYBE_SUDO docker ps | grep 'retool-onpremise_api')"
+api_container_up() {
+  API_DOCKER_CONTAINER = "$($MAYBE_SUDO docker ps | grep 'retool-onpremise-api')"
```
Comment (blackentropy) [inline deploy-retool:222]: why did we change from underscore to hyphen? somewhat concerned about that.

**Review by blackentropy — COMMENTED** [review]
Some quick comments, will try that script out locally now.

File: deploy-retool (hunk 3 above)
Comment (blackentropy) [inline deploy-retool:222]: I think you need to remove the spaces, getting this error:

```/bin/bash: line 222: API_DOCKER_CONTAINER: unbound variable```

Comment (axelchauvin) [reply deploy-retool:222]: because the name shows up as hyphen when I run docker ps. not sure why it was underscore in the first place actually

Comment (axelchauvin) [reply deploy-retool:222]: oops formatting habits. Fixing this

Comment (blackentropy) [reply deploy-retool:222]: shows up as an underscore for me... maybe we should make it a wild card character to be safe? 


image

Comment (blackentropy) [reply deploy-retool:222]: i think different docker versions will format it differently 🤔

File: deploy-retool [hunk 4]
```diff
@@ -242,13 +246,36 @@ fi
...
 
+if ! api_container_up; then
+  echo "ERROR: Failed starting the Retool api container"
+  exit 1
+else
+  API_CONTAINER_ID=$($MAYBE_SUDO docker ps -a | grep -i retool-onpremise[-_]api | awk '{print $1}')
+fi
+
+WAITED=0
+# one request every three seconds. It takes about 40 seconds for the api to get ready so 1 minute should be enough margin
```
Comment (blackentropy) [inline deploy-retool:264]: update the comment!

**Review by blackentropy — APPROVED** [review]
Just one last thing -- but looks great!

**Pull request merged**